deploy/deploy-fifo-market.ts

```diff
@@ -33,9 +38,22 @@ export const deploy: DeployFunction = async (environment) => {
     await certificate.grantRole(
       await certificate.MINTER_ROLE(),
       contract.address
-    ); // todo stop doing this during deployment for cypress tests (use run('nori mint ...') in tests instead)
+    );
   }
   hre.trace('Added FIFOMarket as a minter of Certificate');
+
+  const rNori = await getRestrictedNORI({ hre, signer });
+  const removal = await getRemoval({ hre, signer });
+  const bpNori = await getBridgedPolygonNori({ hre, signer });
+  await rNori.registerContractAddresses(
+    contract.address,
+    bpNori.address,
+    removal.address
+  );
+  hre.trace('Set market, removal and bpNori addresses in rNori');
+  await removal.registerRestrictedNORIAddress(rNori.address);
+  hre.trace('Set rNori address in Removal');
```

Comment:
these seem out of place-- should they live in the restricted nori and removal deploy scripts instead?

Reply:
so setting the addresses in these contracts can't happen until the market contract is already deployed, because `rNori` needs the market contract address and `Removal` and `rNori` need each other's addresses (so one or the other needs to come first before these functions can be called).  At the time of the deploy scripts running for `Removal` and `rNori`, we're not ready to call these functions, so instead we call them after the market is deployed.

Reply:
Right-- I think they all quite fit the scope of "deploy-fifo-market" since they also set config in `rNori` and in `removal`. Perhaps they could live in something like `configure-assets-after-deployment` with the following deploy dependency  `deploy.dependencies = ['market']`

Reply:
good idea.  wrote that, also put the adding of the market contract as a certificate minter in there